scantool/diag_l2_iso14230.c

```diff
@@ -1067,12 +1067,12 @@ dl2p_14230_request(struct diag_l2_conn *d_l2_conn, struct diag_msg *msg,
 		d_l2_conn->diag_msg = NULL;
 
 		/* Check for negative response */
-		if (rmsg->data[0] != DIAG_KW2K_RC_NR) {
+		if (rmsg && rmsg->data[0] != DIAG_KW2K_RC_NR) {
```

Comment:
I remember the cppcheck warning for this, and it's because it can't know that if  _int_recv() is succesful, the messages are added on rmsg. I never really liked that mechanism, but this is an acceptable fix to silence the warning.
Could also be an `assert(rmsg)` once,  to avoid having to check rmsg on all 3 `if` blocks.